### ISSUE TITLE
Runtime error: failed to load jnivideo.so when ffmpeg running the SDK2.1 environment

### ISSUE THREAD

**Issue by Seasky** [havlenapetr/FFMpeg#2]
I want to know the required SDK version for ffmpeg, I saw a runtime error in SDK2.1 ENV. that is:

I/dalvikvm(  223): Unable to dlopen(/data/data/cz.havlena.ffmpeg.ui/lib/libjnivideo.so): Cannot load library: link_image[1721]:    30 could not load needed library 'libsurfaceflinger_client.so' for 'libjnivideo.so' (load_library[1051]: Library 'libsurfaceflinger_client.so' not found)
D/FFMpeg  (  223): Couldn't load lib: jnivideo - Library jnivideo not found

I check the /system/lib/ directory, and do not find the required libsurfaceflinger_client.so, So I must update my SDK to 2.2 or higher,right? If I want ffmpeg compatible with 2.1 platform, how to do it?

**Comment by havlenapetr** [comment]
when you are building apk from your java code and native libs in lib/armeabi you must have my libjnivideo.so and libjniaudio.so too in lib/armeabi, if you haven't them there eclipse doesn't pack them into apk and dalvik can't find them because they doesn't exists:-)

**Comment by Seasky** [comment]
I have put these two files under lib/armeabi, and the jniaudio.so is loaded successfully, but the jnivideo.so failed due to the jnivideo.so need the libsurfaceflinger_client.so, which doesn't exist in SDK 2.1 system(there is no such file in /system/lib), So I ask which is the required SDK version for ffmpeg.

**Comment by havlenapetr** [comment]
libsurfaceflinger_client.so is in froyo but not in eclair, you should recompile my libjnivideo.so http://github.com/havlenapetr/android_frameworks_base/tree/froyo/native/video/jni/ for eclair or if I have time i will do it

**Comment by havlenapetr** [comment]
I think for eclair you just rewrite my makefile(in frameworks/base/native) from libsurfaceflinger_client.so to libsurfaceflinger.so

**Comment by Seasky** [comment]
Thanks for your information. I compared the froyo frameworks/base/native directory with yours, the froyo native directory do not include the /audio and /video sub-directory, So I think these two directories are added by yourself and used to create the libjniaudio.so and libjnivideo.so, right? 
And in android 2.1 platfrom, there aren't native directory under frameworks/base, So do you mean I just need copy the /native directory from your code base, then modify the makefile and can create these two libs too? Does these two libs need other libs or source codes which is not included in 2.1 platform?

**Comment by havlenapetr** [comment]
yes it is my libraries, yes you can compile them for eclair too, but some libraries will not be in eclair, so you must modify my makefiles for froyo(libraries will be there but with other name or other path for headers)

**Comment by yangchaoxun** [comment]
ok？
